test/jdk/java/lang/module/exeNullCallerGetResource/exeNullCallerGetResource.c

```diff
@@ -0,0 +1,188 @@
+    assert(class_Module != NULL);
+    jmethodID mid_Module_getResourceAsStream =
+        (*env)->GetMethodID(env, class_Module, "getResourceAsStream", "(Ljava/lang/String;)Ljava/io/InputStream;" );
+    assert(mid_Module_getResourceAsStream != NULL);
+    jobject in = (*env)->CallObjectMethod(env, n, mid_Module_getResourceAsStream,
+        (*env)->NewStringUTF(env, "open/test.txt"));
+    if ((*env)->ExceptionOccurred(env) != NULL) {
+        printf("ERROR: Exception was thrown calling Module::getResourceAsStream on 'open/test.txt'.\n");
+        (*env)->ExceptionDescribe(env);
+        exit(-1);
+    }
+    assert(in != NULL);
+
+    // in.close();
+    closeInputStream(env, in);
+
+    // in = n.getResourceAsStream("closed/test.txt");
+    in =  (*env)->CallObjectMethod(env, n, mid_Module_getResourceAsStream,
+        (*env)->NewStringUTF(env, "closed/test.txt"));
+    if ((*env)->ExceptionOccurred(env) != NULL) {
+        printf("ERROR: Exception was thrown calling Module::getResourceAsStream on 'closed/test.txt'.\n");
+        (*env)->ExceptionDescribe(env);
+        exit(-1);
+    }
+    assert(in == NULL);
+
+    // in = c.getResourceAsStream("test.txt");
+    jmethodID mid_Class_getResourceAsStream =
+        (*env)->GetMethodID(env, class_Class, "getResourceAsStream", "(Ljava/lang/String;)Ljava/io/InputStream;" );
+    assert(mid_Class_getResourceAsStream != NULL);
+    in =  (*env)->CallObjectMethod(env, c, mid_Class_getResourceAsStream,
+        (*env)->NewStringUTF(env, "test.txt"));
+    if ((*env)->ExceptionOccurred(env) != NULL) {
+        printf("ERROR: Exception was thrown calling Class::getResourceAsStream on 'test.txt'.\n");
+        (*env)->ExceptionDescribe(env);
+        exit(-1);
+    }
+    assert(in != NULL);
+
+    // in.close();
+    closeInputStream(env, in);
+
+    // Class closed = closed.ClosedResources.fetchClass();
+    jclass class_ClosedResources = (*env)->FindClass(env, "closed/ClosedResources");
+    assert(class_ClosedResources != NULL);
+    jmethodID mid_ClosedResources_fetchClass = (*env)->GetStaticMethodID(env, class_ClosedResources, "fetchClass", "()Ljava/lang/Class;" );
+    assert(mid_ClosedResources_fetchClass != NULL);
+    jobject closed =(*env)->CallStaticObjectMethod(env, class_ClosedResources, mid_ClosedResources_fetchClass);
+    if ((*env)->ExceptionOccurred(env) != NULL) {
+        printf("ERROR: Exception was thrown calling ClosedResources::fetchClass.\n");
+        (*env)->ExceptionDescribe(env);
+        exit(-1);
+    }
+    assert(closed != NULL);
+
+    // in = closed.getResourceAsStream("test.txt");
+    in =  (*env)->CallObjectMethod(env, closed, mid_Class_getResourceAsStream,
+        (*env)->NewStringUTF(env, "test.txt"));
+    if ((*env)->ExceptionOccurred(env) != NULL) {
+        printf("ERROR: Exception was thrown calling Class::getResourceAsStream on closed 'test.txt'.\n");
+        (*env)->ExceptionDescribe(env);
+        exit(-1);
+    }
+    assert(in == NULL);
```

Comment:
assert is typically used for sanity test.   As part of the test validation, e.g. in this case `in == NULL` or `in != NULL` in line 157,  it may be clearer if it's an explicit check and throw exception to indicate test failure especially in case `#undef NDEBUG` may not be set in the test.